### PR TITLE
Sof 2265/fetch video mixer configuration

### PR DESCRIPTION
**Pull request by LindvedKrvang**
In order for AlbaTv2Server to connect to the configured VideoMixer of the system, we need to expose the VideoMixer configuraiton (hostname and port).

Since this information is currently is handled by Core (and we don't have time/haven't gotten to point where AlbaServer should handle it) we are doing a less than optimal approach.

This PR introduce an endpoint to fetch this information and a new emit event that emits whenever it is changed.

Once AlbaServer handles Devices this should be deleted again.

### CODE REVIEW

File: src/business-logic/services/interfaces/device-event-emitter.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import {Device} from '../../../model/entities/device'
```
Comment (RasmusAlbrektsen) [inline src/business-logic/services/interfaces/device-event-emitter.ts]: Oh dear god we have inconsistent brackets over here oh no.

**Review by RasmusAlbrektsen — APPROVED** [review]

File: src/business-logic/services/interfaces/device-event-emitter.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 
 export interface DeviceEventEmitter {
   emitDeviceCreatedEvent(device: Device): void
-
   emitDeviceUpdatedEvent(device: Device): void
-
   emitDeviceDeletedEvent(deviceId: string): void
+
+  emitVideoMixerConfigurationUpdated(videoMixerConfiguration: VideoMixerConfiguration): void
 }
```
Comment (MadsThy) [inline src/business-logic/services/interfaces/device-event-emitter.ts]:
```suggestion
import { Device } from '../../../model/entities/device'
import { VideoMixerConfiguration } from '../../../model/value-objects/video-mixer-configuration'

export interface DeviceEventEmitter {
  emitDeviceCreatedEvent(device: Device): void
  emitDeviceUpdatedEvent(device: Device): void
  emitDeviceDeletedEvent(deviceId: string): void
  emitVideoMixerConfigurationUpdated(videoMixerConfiguration: VideoMixerConfiguration): void
}
```

File: src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      options: {
+        host: string
+        port: number
+      }
+    }>
+  }
+}
+
+const COLLECTION_NAME: string = 'peripheralDevices'
+
```
Comment (MadsThy) [inline src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts]: why the spacing? :)

File: src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+const PLAYOUT_GATEWAY_POSTFIX: string = 'PlayoutCoreParent'
+const VIDEO_MIXER_NAME: string = 'atem'
+
+/**
+ * This Repository only exist to fetch the hostname and port of the configured VideoMixer (currently hardcoded to Atem) while device settings resides in Sofie.
+ * Once proper device settings is introduced in Alba. This SHOULD be deleted!
```
Comment (MadsThy) [inline src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts]:
```suggestion
 * This Repository only exists to fetch the hostname and port of the configured VideoMixer (currently hardcoded to Atem) while device settings reside in Sofie.
 * Once proper device settings are introduced in Alba this SHOULD be deleted!
```

File: src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        default: {
+          // Ignore the rest
+        }
+      }
+    })
+  }
+
+  private findVideoMixerConfiguration(mongoDevice: MongoDevice): VideoMixerConfiguration {
+    const key: string | undefined = Object.keys(mongoDevice.settings.devices).find(key => key.includes(VIDEO_MIXER_NAME))
```
Comment (MadsThy) [inline src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts]: From what you are using it for, i would suggest this renaming :)

```suggestion
    const deviceName: string | undefined = Object.keys(mongoDevice.settings.devices).find(key => key.includes(VIDEO_MIXER_NAME))
```

**Review by MadsThy — APPROVED** [review]

File: src/business-logic/services/interfaces/device-event-emitter.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/business-logic/services/interfaces/device-event-emitter.ts]: I did leave the space deliberately to show that there is a semantic grouping

File: src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts (hunk 3 above)
Comment (LindvedKrvang) [reply src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts]: It was an attempt to make a semantic grouping, but I guess that for the constants in questions there is no real grouping in the latter two...

File: src/business-logic/services/interfaces/device-event-emitter.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/business-logic/services/interfaces/device-event-emitter.ts]: Not anymore

File: src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts (hunk 3 above)
Comment (LindvedKrvang) [reply src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts]: The space has been removed

File: src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts (hunk 4 above)
Comment (LindvedKrvang) [reply src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts]: Updated

File: src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts (hunk 5 above)
Comment (LindvedKrvang) [reply src/data-access/repositories/mongo/mongo-video-mixer-device-repository.ts]: Updated

**Review by KvelaGorrrrnio — APPROVED** [review]

**Pull request merged**